### PR TITLE
Bug 989

### LINKED ISSUE #989: [BUG]: Fix id-token cmd + other cmds so that it runs thru mamba
### Operating system

MacOS Ventura 13.2.1

### Bug description

Running the id-token command doesn't work unless the mamba env is spun up already with `mamba activate dlp`. This can be fixed by adding `mamba run` to the commands in the dlp-cli. I likely forgot to do this with other commands if the id-token command doesn't do this automatically, so check to make sure that those commands are fixed also

### Steps to reproduce

Run the id-token command without the mamba environment activated.

### PR DESCRIPTION
**Pull request by yebyyy**
### Bug-989

Github Issue Number Here: https://github.com/DSGT-DLP/Deep-Learning-Playground/issues/989
Ensure dlp-cli Commands Run Within Mamba Environment

modifying uid.go and id-token.go in backend folder

runnning mamba deactivate and go run main.go backend id-token <email here> to test if the cmd can run without mamba activated manually

### CODE REVIEW

File: cmd/backend/id_token/id_token.go [hunk 1]
```diff
@@ -16,10 +16,12 @@ var IdTokenCmd = &cobra.Command{
...
 	Run: func(cmd *cobra.Command, args []string) {
-		backend.ExecBashCmd("poetry", "run", "python", "cli.py", "get-id-token", args[0])
+		env_name := cmd.Flag("env-name").Value.String()
+		backend.ExecBashCmd("mamba", "run", "-n", env_name, "poetry", "run", "python", "cli.py", "get-id-token", args[0])
 	},
 }
 
 func init() {
 	backend.BackendCmd.AddCommand(IdTokenCmd)
+	IdTokenCmd.PersistentFlags().StringP("env-name", "e", "dlp", "The name of the mamba environment")
```
Comment (karkir0003) [inline cmd/backend/id_token/id_token.go]: unneeded

Comment (karkir0003) [reply cmd/backend/id_token/id_token.go]: the `backend.go` file already has the persistent flag of `env-name = "dlp"`

File: cmd/backend/start/start.go [hunk 2]
```diff
@@ -25,4 +25,5 @@ var StartCmd = &cobra.Command{
 func init() {
 	backend.BackendCmd.AddCommand(StartCmd)
 	StartCmd.PersistentFlags().IntP("port", "p", 8000, "A port to run the backend on")
+	StartCmd.PersistentFlags().StringP("env-name", "e", "dlp", "The name of the mamba environment")
```
Comment (karkir0003) [inline cmd/backend/start/start.go]: same as above

**Review by karkir0003 — COMMENTED** [review]
added a few comments. good work on this so far

**Comment by karkir0003** [comment]
update PR description to mention that you also updated the `uid.go` file as well within the available backend commands too

**Review by karkir0003 — COMMENTED** [review]
Looks good to me. Let's have @dwu359 review it

**Review by karkir0003 — APPROVED** [review]

**Comment by karkir0003** [comment]
@yebyyy approved on my end. lets wait on @dwu359 's review

**Review by dwu359 — COMMENTED** [review]
Not needed for this pr, but in case if a bash cmd needs user input in general, you may see a --live-stream flag in other dlp-cli go files that allows for that: [https://docs.conda.io/projects/conda/en/latest/commands/run.html](https://docs.conda.io/projects/conda/en/latest/commands/run.html). I will not review this PR any further, please wait until a currently active project lead approves before merging (I need them to also familiarize themselves with the dlp-cli)

**Comment by alantao912** [comment]
Looks good to me!

**Comment by karkir0003** [comment]
@alantao912 you have to click "approve" on the PR

**Review by alantao912 — APPROVED** [review]
Looks good to me!

**Pull request merged**